File: examples/app-redux-android-todolist-kotlin/src/main/java/com/redux/Operations.java
package com.redux;

import rx.Observable;
import rx.Subscriber;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

import javax.inject.Inject;
import java.io.IOException;
import java.util.List;

public class Operations {

    private final Service service;

    @Inject
    public Operations(Service service) {
        this.service = service;
    }

    Observable<List<Todo>> fetch() {
        return Observable
                .create(new Observable.OnSubscribe<List<Todo>>() {
                    @Override public void call(Subscriber<? super List<Todo>> subscriber) {
                        try {
                            subscriber.onNext(service.get());
                        } catch (IOException | InterruptedException e) {
                            subscriber.onError(e);
                        }
                        subscriber.onCompleted();
                    }
                })
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread());
    }
}

File: examples/app-redux-android-todolist-kotlin/src/main/kotlin/com/redux/Operations.kt
package com.redux

import rx.Observable
import rx.Subscriber
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

import javax.inject.Inject
import java.io.IOException

public class Operations @Inject constructor(private val service: Service) {

    fun fetch(): Observable<List<Todo>> {
        return Observable.create(object : Observable.OnSubscribe<List<Todo>> {
            override fun call(subscriber: Subscriber<in List<Todo>>) {
                try {
                    subscriber.onNext(service.get())
                } catch (e: IOException) {
                    subscriber.onError(e)
                } catch (e: InterruptedException) {
                    subscriber.onError(e)
                }

                subscriber.onCompleted()
            }
        }).subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread())
    }
}
